src/fcntl.rs

```diff
@@ -99,30 +113,37 @@ libc_bitflags!(
                   target_os = "netbsd",
                   target_os = "openbsd",
                   target_os = "redox"))]
+        #[cfg_attr(docsrs, doc(cfg(all())))]
         O_FSYNC;
         /// Allow files whose sizes can't be represented in an `off_t` to be opened.
         #[cfg(any(target_os = "android", target_os = "linux"))]
+        #[cfg_attr(docsrs, doc(cfg(all())))]
         O_LARGEFILE;
         /// Do not update the file last access time during `read(2)`s.
         #[cfg(any(target_os = "android", target_os = "linux"))]
+        #[cfg_attr(docsrs, doc(cfg(all())))]
```

Comment:
Hi @asomers, looking at our source code, I am curious why this `#[cfg_attr(docsrs, doc(cfg(all())))]` is needed?

From my obervation:

1. We are not enabling the [`doc_auto_cfg`](https://doc.rust-lang.org/rustdoc/unstable-features.html#doc_auto_cfg-automatically-generate-doccfg) feature, which means we won't get a `#[doc(cfg(xx))]` with `#[cfg(xxx)]`, so no need to overwrite a non-existing value
2. The `doc_cfg` feature does not seem to work with assocaiated item:

    ```rs
        #![feature(doc_cfg)]

    #[cfg(target_os = "linux")]
    #[doc(cfg(target_os = "linux"))]
    pub struct Foo;

    impl Foo {
        #[cfg(target_os = "linux")]
        #[doc(cfg(target_os = "linux"))]
        pub const LINUX_SPECIFIC_CONST: usize = 1;
    }
   ```
  
   Run:

   ```sh
   $ cargo +nightly doc --target x86_64-unknown-linux-gnu --open
   ```

   That `Available on Linux only.` banner will be set for `Foo` only, not for `LINUX_SPECIFIC_CONST`

And some types in our code are partially annocated with this attribute, e.g., `enum Request` defined in `src/sys/ptrace/linux.rs`, which makes me confused

Reply:
I think they're there because I originally developed the PR with the `doc_auto_cfg` feature (see https://github.com/nix-rust/nix/pull/1498/commits/83ea2f34f1010762fc6fd8783d04191623e6abaf).  When that's on, constants like `O_NOCTTY` get a "This is supported on non-Redox only" banner.  Those banners are both annoyingly verbose and deceptive (see the linked commit message for the explanation for why they're deceptive).  But then I must've removed `doc_auto_cfg` before I posted the PR, leaving the `doc(cfg(all()))` attributes useless.  So we can remove them now.